vent/io/devices.py

```diff
@@ -0,0 +1,353 @@
+    '''
+    _DEFAULT_ADDRESS     = 0x40
+    _FLOW_OFFSET         = 32768
+    _FLOW_SCALE_FACTOR   = 120
+
+    def __init__(self, address=_DEFAULT_ADDRESS, i2c_bus=1, pig=None):
+        I2CDevice.__init__(self,address,i2c_bus,pig)
+        Sensor.__init__(self)
+        self.reset()
+        self._start()
+
+    def reset(self):
+        ''' Extended to add device specific behavior: Asks the sensor
+        to perform a soft reset. 80 ms soft reset time.
+        '''
+        super().reset()
+        self.write_device(0x2000)
+        sleep(.08)
+
+    def _start(self):
+        ''' Device specific:Sends the 'start measurement' command to the
+        sensor. Start-up time once command has been recieved is
+        'less than 100ms'
+        '''
+        self.write_device(0x1000)
+        sleep(.1)
+
+    def _verify(self,value):
+        ''' No further verification needed for this sensor. Onboard
+        chip handles all that. Could throw in a CRC8 checker instead of
+        discarding them in _convert().
+        '''
+        return True
+
+    def _convert(self,raw):
+        ''' Overloaded to replace with device-specific protocol.
+        Convert raw int to a flow reading having type float with
+        units slm. Implementation differs from parent for clarity and
+        consistency with source material.
+
+        Source:
+          https://www.sensirion.com/fileadmin/user_upload/customers/sensirion/Dokumente/ ...
+            ... 5_Mass_Flow_Meters/Application_Notes/Sensirion_Mass_Flo_Meters_SFM3xxx_I2C_Functional_Description.pdf
+        '''
+        return (raw - self._FLOW_OFFSET) / self._FLOW_SCALE_FACTOR
+
+    def _raw_read(self):
+        ''' Performs an read on the sensor, converts recieved bytearray,
+        discards the last two bytes (crc values - could implement in future),
+        and returns a signed int converted from the big endian two
+        complement that remains.
+        '''
+        return be16_to_native(self.read_device(4)[:2])
+
+
+#class HumiditySensor(Sensor):
+#    ''' Not yet implemented.
+#    '''
+#    def __init__(self):
+#        raise NotImplementedError
+
+
+#class TemperatureSensor(Sensor, SPIDevice):
+#    ''' Not yet implemented
+#    '''
+#    def __init__(self):
+#        raise NotImplementedError
+
+
+class SolenoidValve(OutputPin):
+    ''' An extension of OutputPin which uses valve terminology for its
+    methods. Also allows configuring both normally open and normally
+    closed valves (called the "form" of the valve).
+    '''
+    _FORMS = {  'Normally Closed'   : 0,
+                'Normally Open'     : 1 }
+    def __init__( self, pin, form='Normally Closed', pig=None ):
+        self.form = self._FORMS[form]
+        super().__init__(pin,pig)
+
+    @property
+    def form(self):
+        ''' Returns the human-readable form of the valve
+        '''
+        return dict(map(reversed,self._FORMS.items()))[self._form]
+
+    @form.setter
```

Comment:
It would be neat to use the same style consistently. @property and
   @....setter look good and ".form" is easy to use. Across modules, set_ and get_ might be more explicit though.